src/napari/_qt/layer_controls/widgets/_labels/qt_label_color.py

```diff
@@ -0,0 +1,188 @@
+            self.color = tuple(color)
+
+    def disconnect_widget_controls(self) -> None:
+        disconnect_events(self._layer.events, self)
+
+    def deleteLater(self) -> None:
+        self.disconnect_widget_controls()
+        super().deleteLater()
+
+    def closeEvent(self, event) -> None:
+        """Disconnect events when widget is closing."""
+        self.disconnect_widget_controls()
+        super().closeEvent(event)
+
+
+class QtLabelControl(QtWidgetControlsBase):
+    """
+    Class that wraps the connection of events/signals between the current label
+    layer attribute and Qt widgets.
+
+    Parameters
+    ----------
+    parent: qtpy.QtWidgets.QWidget
+        An instance of QWidget that will be used as widgets parent
+    layer : napari.layers.Layer
+        An instance of a napari layer.
+
+    Attributes
+    ----------
+    colorbox : QtColorBox
+        Widget that shows current layer label color.
+    label_color : qtpy.QtWidget.QWidget
+        Wrapper widget for the selection_spinbox and colorbox widgets.
+    label_color_label : napari._qt.layer_controls.widgets.qt_widget_controls_base.QtWrappedLabel
+        Label for the label chooser widget.
+    selection_spinbox : superqt.QLargeIntSpinBox
+        Widget to select a specific label by its index.
+        N.B. cannot represent labels > 2**53.
+    """
+
+    def __init__(
+        self, parent: QWidget, layer: Layer, tooltip: Optional[str] = None
+    ) -> None:
+        super().__init__(parent, layer)
+        # Setup layer
+        self._layer.events.selected_label.connect(
+            self._on_selected_label_change
+        )
+        self._layer.events.data.connect(self._on_data_change)
+
+        # Setup widgets
+        self.selection_spinbox = QLargeIntSpinBox()
+        dtype_lims = get_dtype_limits(get_dtype(layer))
+        self.selection_spinbox.setRange(*dtype_lims)
+        self.selection_spinbox.setKeyboardTracking(False)
+        self.selection_spinbox.valueChanged.connect(self.change_selection)
+        self.selection_spinbox.setAlignment(Qt.AlignmentFlag.AlignCenter)
+        self._on_selected_label_change()
+
+        self.label_color_label = QtWrappedLabel(trans._('label:'))
+        self.label_color = QWidget()
+        self.label_color.setProperty('emphasized', True)
+        color_layout = QHBoxLayout()
+        color_layout.setContentsMargins(0, 2, 0, 1)
+        self.colorbox = QtColorBox(layer)
+        color_layout.addWidget(self.colorbox)
+        color_layout.addWidget(self.selection_spinbox)
+        self.label_color.setLayout(color_layout)
+
+    def change_selection(self, value: int) -> None:
+        """Change currently selected label.
+
+        Parameters
+        ----------
+        value : int
+            Index of label to select.
+        """
+        self._layer.selected_label = value
+        self.selection_spinbox.clearFocus()
+        # TODO: decouple
+        self.parent().setFocus()
```

Comment:
see below about setfocus.

Reply:
Similar to https://github.com/napari/napari/pull/7355#discussion_r2219937658, https://github.com/napari/napari/pull/7355#discussion_r2219788810 and https://github.com/napari/napari/pull/7355#discussion_r2219783030 In the current labels controls widget this looks like

https://github.com/napari/napari/blob/7efa901b72e360f5186d13d3b4c3a7cc21a05a7b/src/napari/_qt/layer_controls/qt_labels_controls.py#L376-L386

Happy to remove this logic if it is not needed anymore